unicodetools/src/main/java/org/unicode/text/UCD/VersionedProperty.java

```diff
@@ -0,0 +1,141 @@
+package org.unicode.text.UCD;
+
+import com.ibm.icu.dev.util.UnicodeMap;
+import com.ibm.icu.text.SymbolTable;
+import com.ibm.icu.text.UnicodeSet;
+import java.text.ParsePosition;
+import java.util.Map;
+import java.util.Set;
+import org.unicode.props.IndexUnicodeProperties;
+import org.unicode.props.UnicodeProperty;
+import org.unicode.props.UnicodeProperty.Factory;
+import org.unicode.props.UnicodeProperty.PatternMatcher;
+import org.unicode.text.utility.Settings;
+
+public class VersionedProperty {
+    private String propertyName;
+    private String version;
+    private UnicodeProperty.Factory propSource;
+    private UnicodeProperty property;
+    private final transient PatternMatcher matcher = new UnicodeProperty.RegexMatcher();
+
+    private static final Set<String> TOOL_ONLY_PROPERTIES =
```

Comment:
BTW, just checked in changes to a PR so that the normalization code can use the IndexUnicodeProperties.